File: pwa/src/sessionId.test.ts
import { describe, expect, it } from 'vitest';
import { nextSessionId } from './sessionId';

describe('nextSessionId', () => {
  it('returns YYYY-MM-DD when no existing ids for that date', () => {
    expect(nextSessionId('2026-05-10', [])).toBe('2026-05-10');
    expect(nextSessionId('2026-05-10', ['2026-05-09'])).toBe('2026-05-10');
  });

  it('appends -2 when one session already exists for the date', () => {
    expect(nextSessionId('2026-05-10', ['2026-05-10'])).toBe('2026-05-10-2');
  });

  it('appends -3 when -2 already exists', () => {
    expect(nextSessionId('2026-05-10', ['2026-05-10', '2026-05-10-2'])).toBe('2026-05-10-3');
  });

  it('handles non-contiguous suffixes (uses max + 1)', () => {
    expect(nextSessionId('2026-05-10', ['2026-05-10', '2026-05-10-5'])).toBe('2026-05-10-6');
  });

  it('ignores ids for other dates', () => {
    expect(nextSessionId('2026-05-10', ['2026-05-09', '2026-05-09-2', '2026-05-11'])).toBe('2026-05-10');
  });
});

File: flutter/test/session_id_test.dart
import 'package:flutter_test/flutter_test.dart';
import 'package:home_hd/features/treatment/session_id.dart';

void main() {
  group('nextSessionId', () {
    test('no existing → bare date', () {
      expect(nextSessionId('2026-06-02', []), '2026-06-02');
      expect(nextSessionId('2026-06-02', ['2026-06-01']), '2026-06-02');
    });

    test('one same-day session → -2', () {
      expect(nextSessionId('2026-06-02', ['2026-06-02']), '2026-06-02-2');
    });

    test('uses max suffix + 1, not count', () {
      expect(
        nextSessionId('2026-06-02', ['2026-06-02', '2026-06-02-3']),
        '2026-06-02-4',
      );
    });

    test('ignores other dates', () {
      expect(
        nextSessionId('2026-06-02', ['2026-06-01-5', '2026-06-02']),
        '2026-06-02-2',
      );
    });
  });

  test('todayIso / nowHHMM zero-pad', () {
    final t = DateTime(2026, 3, 5, 9, 7);
    expect(todayIso(t), '2026-03-05');
    expect(nowHHMM(t), '09:07');
  });
}
